src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -182,32 +182,35 @@
+       This achieves two things: 
+	   1) Allows VS to work when the test dir is initially clean, since VS does not know how to invoke the runner script
+	   2) For incremental build of any kind, ensures that if direct dependencies that also are in the output of 
+	      PrereleaseResolveNuGetPackageAssets are updated, they get copied (thanks to smarter copy logic)
        -->
-  <PropertyGroup Condition="'$(BuildingInsideVisualStudio)'=='true'">
-    <PrepareForRunDependsOn>$(PrepareForRunDependsOn);CopyDefaultTestAssetsForVS</PrepareForRunDependsOn>
+  <PropertyGroup Condition="'$(ArchiveTests)' != 'true'">
+    <PrepareForRunDependsOn>$(PrepareForRunDependsOn);CopyDefaultTestAssets</PrepareForRunDependsOn>
   </PropertyGroup>
-  <Target Name="CopyDefaultTestAssetsForVS" DependsOnTargets="CopyTestToTestDirectory;CopySupplementalTestData">
+  <Target Name="CopyDefaultTestAssets" DependsOnTargets="CopyTestToTestDirectory;CopySupplementalTestData">
      <!-- This was copied from RunTestsForProject in tests.targets
           The RunTestsForProject target does not execute in VS context and would be confused by the script based runner.
           _TestCopyLocalByFileNameWithoutDuplicates are the precise items that are fed to the runner script generation code.
      -->
     <ItemGroup>
       <!-- Not all platforms can use the .ni.dlls that come from packages.  If TestWithoutNativeImages is specified, we'll exclude them from copy generation.
            If we end up needing this for any other sorts of filtering, we'll want to add a list of filtered extensions to be matched on EndsWith.  -->
-      <_IncludedFileForTestsInVS Include="@(_TestCopyLocalByFileNameWithoutDuplicates)"
+      <_IncludedFileForTests Include="@(_TestCopyLocalByFileNameWithoutDuplicates)"
                                  Condition="'$(TestWithoutNativeImages)' != 'true' Or !$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').EndsWith('.ni.dll'))" >
         <PackageRelativePath Condition="'%(_TestCopyLocalByFileNameWithoutDuplicates.NugetPackageId)' != ''">$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').Replace('$(PackagesDir)',''))</PackageRelativePath>
         <UseAbsolutePath Condition="'$(TestWithLocalLibraries)'=='true'">$([System.String]::Copy('%(_TestCopyLocalByFileNameWithoutDuplicates.SourcePath)').StartsWith('$(BinDir)'))</UseAbsolutePath>
-        <DestinationPath>$(TestPath)\%(Filename)%(Extension)</DestinationPath>
-      </_IncludedFileForTestsInVS>
-      <_IncludedFileForTestsInVs Remove="@(_IncludedFileForTestsInVS)" Condition="Exists('%(DestinationPath)')" />
+        <DestinationPath>$(TestPath)$(TestNugetTargetMonikerFolder)\%(Filename)%(Extension)</DestinationPath>
+      </_IncludedFileForTests>
+      <_IncludedFileForTests Remove="@(_IncludedFileForTests)" Condition="Exists('%(DestinationPath)')" />
```

Comment:
>@weshaggard The consolidation of the computation of _IncludedFilesForTests can probably be done in a separate PR, as it's one of the things I'm working on for #1116.

I agree it could be done separately, but I would prefer we not start using msbuild to copy until we have some confidence that it will be copying the same stuff.